File: app/src/main/org-api/models/event.py
"""Event class module"""

from models.org_base_model import BaseModel
from sqlalchemy import Column, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
import models

class Event(BaseModel):
    """ Event model class
    Attributes:
        event_id (str): Primary key for the event.
        event_title (str): The title or name of the event.
        description (str): Description or details about the event.
        attendees (list): List of member IDs attending the event.
        rsvps (list): List of member IDs who have RSVP'd to the event.
        location (str): The location or venue of the event.
        event_date (DateTime): Date and time of the event.
        author_id (str): Foreign key to the member who created the event.

    Relationships:
        created_by (relationship): Relationship to the Member model for the event creator.
    """
    if models.storage_type == "database":
        # Define the table name for the Event model
        __tablename__ = "events"
        event_title = Column(String(128), nullable=False)
        description = Column(String(256))
        location = Column(String(128))
        event_date = Column(DateTime, nullable=False)
        author_id = Column(String(60), ForeignKey('members.member_id'), nullable=False)
        
        # Define relationships
        attendees = relationship('Member', secondary='event_attendees')
        rsvps = relationship('Member', secondary='event_rsvps')
        author = relationship('Member', foreign_keys=[author_id])
    else:
        event_title = ""
        description = ""
        location = ""
        event_date = None
        author_id = ""
        attendees = []
        rsvps = []

    def __init__(self, *args, **kwargs):
        """ Class constructor """
        super().__init__(*args, **kwargs)

File: app/src/main/java/com/abstratsystems/org/models/Event.kt
import com.google.gson.annotations.SerializedName

/**
 * Data class representing an Event in the ORG Android app.
 *
 * @property id Unique identifier for the event.
 * @property eventTitle Title of the event.
 * @property description Description of the event.
 * @property attendees List of member IDs attending the event.
 * @property rsvps List of member IDs who RSVPed for the event.
 * @property location Location of the event.
 * @property eventDate Date and time of the event.
 * @property authorId Identifier of the member who created the event.
 * @property createdAt Timestamp indicating when the event was created.
 * @property updatedAt Timestamp indicating the last update to the event information.
 */
data class Event(
    @SerializedName("id") var id: String?,
    @SerializedName("event_title") val eventTitle: String,
    val description: String,
    val attendees: ArrayList<String>,
    val rsvps: ArrayList<String>,
    val location: String,
    @SerializedName("event_date") val eventDate: String,
    @SerializedName("author_id") val authorId: String,
    @SerializedName("created_at") var createdAt: String?,
    @SerializedName("updated_at") var updatedAt: String?
)
